internal/handlers/common/filter.go

```diff
@@ -720,6 +715,11 @@ func filterFieldExprSize(fieldValue any, sizeValue any) (bool, error) {
 		)
 	}
 
+	arr, ok := fieldValue.(*types.Array)
+	if !ok {
+		return false, nil
+	}
```

Comment:
We check array type last because compat test told us we should check values first.

Reply:
@chilagrow as codecov complains, do we have a test for `size` where we call it for a non-array? Or is it a separate issue?

Reply:
Yes we do, we call them from many non array. Is this a problem?

For `$size` and `$elemMatch` we now call from every provider from shareddata such as [Double](https://github.com/FerretDB/FerretDB/blob/main/integration/shareddata/scalars.go#L101).

Reply:
Sounds reasonable! I wonder why codecov didn't catch it. Anyway, it's not in the scope of the current task.